docs/gatekeeper-policy-authoring.md

```diff
@@ -0,0 +1,254 @@
+the results of all verifiers back to Gatekeeper so that policies can be authored
+in rego.
+
+Rego policies utilizing external data are authored in exactly the same way as
+traditional rego policies, but need to know the structure of the response from
+the external data provider in order to take advantage of it.
+
+## Rego References
+These are some helpful references for working with rego:
+- [Rego Policy Language](https://www.openpolicyagent.org/docs/latest/policy-language/)
+- [Rego Policy Reference](https://www.openpolicyagent.org/docs/latest/policy-reference/)
+- [Rego Built-in Functions](https://www.openpolicyagent.org/docs/latest/policy-reference/#built-in-functions)
+- [Gatekeeper External Data](https://open-policy-agent.github.io/gatekeeper/website/docs/externaldata)
+- [Gatekeeper Policy Debugging](https://open-policy-agent.github.io/gatekeeper/website/docs/debug)
+- [Rego Playground](https://play.openpolicyagent.org/)
+
+The rego playground is a great way to test out policies against responses from
+Ratify. Use the steps from the Gatekeeper debugging reference to print the
+response from Ratify. This can then be used as the input for rego playground. In
+rego playground you can replace the external_data call with the input:
+```rego
+package ratify
+# You would use the below when running the rego using external_data
+remote_data := response {
+  images := [img | img = input.review.object.spec.containers[_].image]
+  response := external_data({"provider": "ratify-provider", "keys": images})
+}
+# Use this to mimic the behavior of external_data within rego playground
+remote_data := response {
+  response := input
+}
+```
+
+## Ratify Response in Rego
+Let's assume we have a registry with the following image and associated
+artifacts:
+```
+localhost:5000/net-monitor:v1
+    - sbom
+    - notaryv2 signature
+    - cosign signature
+```
+
+We will try to deploy the following pod spec:
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: test-pod
+spec:
+  containers:
+    - name: net-monitor
+      image: localhost:5000/net-monitor:v1
+    - name: missing-image
+      image: localhost:5000/missing-image:v1
+```
+Note that one of the images listed in the pod spec does not actually exist in
+the registry. This is to give an example of something that might generate an
+error message in one of the response fields.
+
+Given the following rego snippet:
+```rego
+package ratify
+# Get data from Ratify
+remote_data := response {
+  images := [img | img = input.review.object.spec.containers[_].image]
+  response := external_data({"provider": "ratify-provider", "keys": images})
+}
+```
+
+The `remote_data` document contents would look something like this (NOTE: this
+is a sample response to explain the structure of the response data and the
+actual values of fields may differ):
+```json
+{
+  "errors": [
+    ["localhost:5000/missing-image:v1", "The subject did not exist"]
+  ],
+  "responses": [
+    [
```

Comment:
Would recommend adding some kind of version property, if this payload is going out of the process boundary and is consumed by external parties.

Reply:
Added #261 to track adding a version to the payload.